pubsub/aws/snssqs/snssqs.go

```diff
@@ -329,31 +146,56 @@ func (s *snsSqs) Init(metadata pubsub.Metadata) error {
 	// both Publish and Subscribe need reference the topic ARN, queue ARN and subscription ARN between topic and queue
 	// track these ARNs in these maps.
 	s.topics = sync.Map{}
-	s.topicsSanitized = sync.Map{}
+	s.sanitizedTopics = sync.Map{}
 	s.queues = sync.Map{}
 	s.subscriptions = sync.Map{}
 
 	sess, err := aws_auth.GetClient(md.AccessKey, md.SecretKey, md.SessionToken, md.Region, md.Endpoint)
 	if err != nil {
 		return fmt.Errorf("error creating an AWS client: %w", err)
 	}
-
+	// AWS sns,sqs,sts client.
+	s.snsClient = sns.New(sess)
+	s.sqsClient = sqs.New(sess)
 	s.stsClient = sts.New(sess)
-	callerIDOutput, err := s.stsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+
+	s.opsTimeout = time.Duration(md.assetsManagementTimeoutSeconds * float64(time.Second))
+	s.ctx, s.cancelFn = context.WithCancel(context.Background())
+
+	if err := s.setAwsAccountIDIfNotProvided(); err != nil {
+		return err
+	}
+
+	// Default retry configuration is used if no
+	// backOff properties are set.
+	if err := retry.DecodeConfigWithPrefix(
+		&s.backOffConfig,
+		metadata.Properties,
+		"backOff"); err != nil {
+		return fmt.Errorf("error decoding backOff config: %w", err)
+	}
+
+	return nil
+}
+
+func (s *snsSqs) setAwsAccountIDIfNotProvided() error {
+	if len(s.metadata.accountID) == awsAccountIDLength {
+		return nil
+	}
+
+	ctx, cancelFn := context.WithTimeout(s.ctx, s.opsTimeout)
+	defer cancelFn()
+
+	callerIDOutput, err := s.stsClient.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
 	if err != nil {
 		return fmt.Errorf("error fetching sts caller ID: %w", err)
 	}
 
 	s.metadata.accountID = *callerIDOutput.Account
-
-	s.snsClient = sns.New(sess)
-	s.sqsClient = sqs.New(sess)
-
 	return nil
 }
 
 func (s *snsSqs) buildARN(serviceName, entityName string) string {
-	// arn:aws:sns:us-east-1:302212680347:aws-controltower-SecurityNotifications
```

Comment:
Keep an example or a doc link is useful.

Reply:
this was in the source by mistake (with my private aws account id oh dear) and therefore removed, besides that it is internal and builds a standard AWS ARN so I don't think it needs documentation